src/lib/ui/SupportedDevices/SupportedDevicesFilterArea.tsx

```diff
@@ -0,0 +1,163 @@
+import { capitalize } from '@mui/material'
+import type { Dispatch, SetStateAction } from 'react'
+
+import { CloseIcon } from 'lib/icons/Close.js'
+import { SearchIcon } from 'lib/icons/Search.js'
+import { Button } from 'lib/ui/Button.js'
+import Menu from 'lib/ui/Menu/Menu.js'
+import { FilterCategoryMenu } from 'lib/ui/SupportedDevices/FilterCategoryMenu.js'
+import type { DeviceModel, Filters } from 'lib/ui/SupportedDevices/types.js'
+import { TextField } from 'lib/ui/TextField/TextField.js'
+
+interface SupportedDevicesFilterAreaProps {
+  deviceModels: DeviceModel[]
+  filterStr: string
+  setFilterStr: (filterStr: string) => void
+  filters: Filters
+  setFilters: Dispatch<SetStateAction<Filters>>
+}
+
+export function SupportedDevicesFilterArea({
+  deviceModels,
+  filterStr,
+  setFilterStr,
+  filters,
+  setFilters,
+}: SupportedDevicesFilterAreaProps) {
+  const appliedFilters = Object.keys(filters).filter((key) => {
+    return (
+      filters[key as keyof Filters] !== undefined &&
+      filters[key as keyof Filters] !== null &&
+      filters[key as keyof Filters] !== false
+    )
+  })
```

Comment:
Does this work?
```suggestion
  const appliedFilters = Object.keys(filters).filter((key: keyof Filters) => {
    return (
      filters[key] != null &&
      filters[key] !== false
    )
  })
```

Reply:
Not easily, no. I'll assign the casted value to a variable to reduce repetition, though.